src/components/views/rooms/PinnedEventTile.tsx

```diff
@@ -52,6 +56,47 @@ export default class PinnedEventTile extends React.Component<IProps> {
         });
     };
 
+    // For event types like polls that use relations, we fetch those manually on
+    // mount and store them here, exposing them through getRelationsForEvent
+    private relations = new Map<string, Map<string, Relations>>();
+    private getRelationsForEvent = (eventId: string, relationType: string, eventType: string): Relations => {
+        if (eventId === this.props.event.getId()) {
+            return this.relations.get(relationType)?.get(eventType);
+        }
+    };
+
+    async componentDidMount() {
+        // Fetch poll responses
+        if (M_POLL_START.matches(this.props.event.getType())) {
+            const eventId = this.props.event.getId();
+            const roomId = this.props.event.getRoomId();
+            const room = this.context.getRoom(roomId);
+
+            try {
+                await Promise.all(
+                    [M_POLL_RESPONSE.name, M_POLL_RESPONSE.altName, M_POLL_END.name, M_POLL_END.altName]
```

Comment:
Rather than having to enumerate all the possible event types that can reference a poll start you can use the bundled relationship.

Looking at a poll event source, you'll see. Fetching all the chunks listed here should give you a comprehensive view of a poll and its responses

```jsonc
{
  "type": "org.matrix.msc3381.poll.start",
  // ...
  "content": {
    // ...
  },
  "origin_server_ts": 1,
  "unsigned": {
    "age": 1,
    "m.relations": {
      "m.reference": {
        "chunk": [
          {
            "event_id": "$hJMr8vQFUVOrvnvHOv6X1hMh0vQE9jBroewXkQTHTAo"
          }
          // ...
        ],
        "next_batch": "t38-2062_0_0_0_0_0_0_0_0"
      }
    }
  },
  "event_id": "$event123",
  "user_id": "@alice:matrix.org",
  "age": 2324572194
}
```

Reply:
Also wondering whether this should live on the JS sdk level, as there are many event types that could face the same issue